src/command/command.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import EmitterMixin from '../emittermixin.js';
+import utils from '../utils.js';
+
+/**
+ * The base class for CKEditor commands.
+ *
+ * Commands are main way to manipulate editor contents and state. They are mostly used by UI elements (or by other
+ * commands) to make changes in Tree Model. Commands are available in every part of code that has access to
+ * {@link core.Editor} instance, since they are registered in it and executed through {@link core.Editor#execute}.
+ * Commands instances are available through {@link core.Editor#commands}.
+ *
+ * This is an abstract base class for all commands.
+ *
+ * @memberOf core.command
+ */
+export default class Command {
+	/**
+	 * Creates a new Command instance.
+	 *
+	 * @param {core.Editor} editor Editor on which this command will be used.
+	 */
+	constructor( editor ) {
+		/**
+		 * Editor on which this command will be used.
+		 *
+		 * @member {core.Editor} core.command.Command#editor
+		 */
+		this.editor = editor;
+
+		/**
+		 * Flag indicating whether a command is enabled or disabled.
+		 * A disabled command should do nothing upon it's execution.
+		 *
+		 * @member {Boolean} core.command.Command#isEnabled
+		 */
+		this.isEnabled = true;
```

Comment:
`Command#isEnabled` and `Command#value` must be observable. Otherwise, there will be no easy way to create e.g. a button's model based on an existing command.

So:

1. `Command` must implement `utils.Observable`.
2. `Command#value` must not be a getter, but a property which is automatically updated once the value is changed. In case of the attribute command that can be implemented as a selection's observer. On selection changes, check the value of the command.